File: 1 Python/Codeforces-686A-FreeIcecream.py
n,x = map(int, input().rstrip().split())
icecreams = x
distressed = 0

for i in range(n):
    inpL = list(map(str,input().rstrip().split()))
    if inpL[0]=="+":
        icecreams+=int(inpL[1])
    else:
        if int(inpL[1])<=icecreams:
            icecreams-=int(inpL[1])
        else:
            distressed+=1

ans = [icecreams, distressed]

print(*ans)

File: 2 CPP/Codeforces-686A-FreeIcecream.cpp
#include <iostream>
using namespace std;

int main(){
    int n;
    long long int x;
    cin >> n >> x;
    long long int  icecreams = x;
    int  distressed = 0;
    
    while(n--){
        char op;
        long long int magnitude;
        cin >> op >> magnitude;
        if(op=='+'){
            icecreams+=magnitude;
        }
        else{
            if(magnitude<=icecreams){
                icecreams-=magnitude;
            }
            else{
                distressed+=1;
            }
        }
    }
    cout << icecreams <<" "<<distressed << "\n";
}
